### ISSUE TITLE
withItems incorrect type in full CRD

### ISSUE THREAD

**Issue by alexec** [argoproj/argo#3578]
Is

```yaml
                            withItems:
                              items:
                                properties:
                                  value:
                                    format: byte
                                    type: string
                                required:
                                - value
                                type: object
                              type: array
```

Should be:

```yaml
                            withItems:
                              items:
                                type: object
                              type: array
```

**Comment by alexec** [comment]
It is also wrong for parallel steps.

**Comment by alexec** [comment]
https://github.com/kubernetes-sigs/controller-tools/issues/391

**Comment by alexec** [comment]
https://github.com/kubernetes-sigs/controller-tools/issues/301

**Comment by alexec** [comment]
Wrong issue linked.

**Comment by alexec** [comment]
Available for testing in [v2.10.0-rc1](https://github.com/argoproj/argo/releases/tag/v2.11.0-rc1).
